### ISSUE TITLE
[WebSocket] newFrameSignal access is not thread safe

### ISSUE THREAD

**Issue by discodonut** [Benedicht/BestHTTP-Issues#14]
When I hit the End Play button in the Unity editor while I have an active web socket connection, I sometimes (maybe 50% of the time?) get one of two different errors:
```
ObjectDisposedException: Safe handle has been closed
System.Runtime.InteropServices.SafeHandle.DangerousAddRef (System.Boolean& success) (at <437ba245d8404784b9fbab9b439ac908>:0)
System.Threading.NativeEventCalls.SetEvent (Microsoft.Win32.SafeHandles.SafeWaitHandle handle) (at <437ba245d8404784b9fbab9b439ac908>:0)
System.Threading.EventWaitHandle.Set () (at <437ba245d8404784b9fbab9b439ac908>:0)
(wrapper remoting-invoke-with-check) System.Threading.EventWaitHandle.Set()
BestHTTP.WebSocket.WebSocketResponse.CloseWithError (BestHTTP.HTTPRequestStates state, System.String message) (at Assets/External/Runtime/Best HTTP/Source/WebSocket/WebSocketResponse.cs:658)
BestHTTP.WebSocket.WebSocketResponse.BestHTTP.Core.IProtocol.CancellationRequested () (at Assets/External/Runtime/Best HTTP/Source/WebSocket/WebSocketResponse.cs:678)
BestHTTP.Core.ProtocolEventHelper.CancelActiveProtocols () (at Assets/External/Runtime/Best HTTP/Source/Core/ProtocolEvents.cs:95)
BestHTTP.HTTPManager.AbortAll () (at Assets/External/Runtime/Best HTTP/Source/HTTPManager.cs:416)
BestHTTP.HTTPManager.OnQuit () (at Assets/External/Runtime/Best HTTP/Source/HTTPManager.cs:388)
BestHTTP.HTTPUpdateDelegator.OnApplicationQuit () (at Assets/External/Runtime/Best HTTP/Source/HTTPUpdateDelegator.cs:232)
```
or
```
NullReferenceException: Object reference not set to an instance of an object
BestHTTP.WebSocket.WebSocketResponse.CloseWithError (BestHTTP.HTTPRequestStates state, System.String message) (at Assets/External/Runtime/Best HTTP/Source/WebSocket/WebSocketResponse.cs:658)
BestHTTP.WebSocket.WebSocketResponse.BestHTTP.Core.IProtocol.CancellationRequested () (at Assets/External/Runtime/Best HTTP/Source/WebSocket/WebSocketResponse.cs:678)
BestHTTP.Core.ProtocolEventHelper.CancelActiveProtocols () (at Assets/External/Runtime/Best HTTP/Source/Core/ProtocolEvents.cs:95)
BestHTTP.HTTPManager.AbortAll () (at Assets/External/Runtime/Best HTTP/Source/HTTPManager.cs:416)
BestHTTP.HTTPManager.OnQuit () (at Assets/External/Runtime/Best HTTP/Source/HTTPManager.cs:388)
BestHTTP.HTTPUpdateDelegator.OnApplicationQuit () (at Assets/External/Runtime/Best HTTP/Source/HTTPUpdateDelegator.cs:232)
```

It looks like this is just a simple race condition between the background thread disposing and nulling `newFrameSignal` and the main thread trying to set it.
```
BestHTTP.WebSocket.WebSocketResponse:ReceiveThreadFunc() (at Assets/External/Runtime/Best HTTP/Source/WebSocket/WebSocketResponse.cs:508)
BestHTTP.PlatformSupport.Threading.<>c__DisplayClass4_0:<RunLongLiving>b__0() (at Assets/External/Runtime/Best HTTP/Source/PlatformSupport/Threading/ThreadedRunner.cs:79)
System.Threading.ThreadHelper:ThreadStart(Object)
```

**Comment by Benedicht** [comment]
What version of the plugn are you using?

**Comment by discodonut** [comment]
We were using a version that you sent me directly on Mar 17, I've updated to the latest available on the asset store (2.0.6?) and I'm still getting the same bug.
